analysis/import/SonarImporter/src/main/kotlin/de/maibornwolff/codecharta/importer/sonar/SonarImporterMain.kt

```diff
@@ -93,4 +95,26 @@ class SonarImporterMain(
     }
 
     override fun getDialog(): ParserDialogInterface = ParserDialog
+    override fun isUsable(inputFile: String): Boolean {
+        val trimmedInput = inputFile.trim()
+        // TODO: Is this sufficient to check for urls?
+        if (trimmedInput.startsWith("http://") || trimmedInput.startsWith("https://")) {
+            return true
+        }
+
+        if (trimmedInput.endsWith("sonar-project.properties")) {
+            return true
+        }
+
+        val searchFile = File(inputFile)
+        val files = searchFile.walk().asSequence()
+                .filter { it.isFile }
+                .map { it.name }
```

Comment:
We could shorten this to 
```suggestion
return trimmedInput.startsWith("http://") || trimmedInput.startsWith("https://") || trimmedInput.endsWith("sonar-project.properties") || searchFile.walk().asSequence()
                .filter { it.isFile }
                .any{ it.name == "sonar-project.properties" }
```

Reply:
Sure we could do that - although I find it a bit ugly to look at :D